### PR TITLE
[#7256] Unit tests and other tweaks to new API endpoint (main)

### PR DESCRIPTION
**Pull request by korydraughn**
The unit test passes at the bench.

Will check the documentation to verify it is rendered correctly.

### CODE REVIEW

File: lib/api/include/irods/get_resource_info_for_operation.h [hunk 1]
```diff
@@ -9,56 +10,67 @@ struct RcComm;
...
-/// char* char_buffer = 0;
-/// addKeyVal(&input.condInput, "GET_RESOURCE_INFO_OP_TYPE_KW", "CREATE");
-/// addKeyVal(&input.condInput, "RESC_NAME_KW", "my_root_resource");
-/// const int ec = rc_get_resource_info_for_operation(&comm, &inp, &char_buffer);
+/// char* char_buffer = NULL;
+///
+/// addKeyVal(&inp.condInput, "GET_RESOURCE_INFO_OP_TYPE_KW", "CREATE");
+/// addKeyVal(&inp.condInput, "RESC_NAME_KW", "my_root_resource");
+///
+/// const int ec = rs_get_resource_info_for_operation(comm, &inp, &char_buffer);
```
Comment (korydraughn) [inline lib/api/include/irods/get_resource_info_for_operation.h:57]: Add more words explaining how to interpret these.

_Apply to other header file._

File: server/api/src/rs_get_resource_info_for_operation.cpp [hunk 2]
```diff
@@ -1,34 +1,39 @@
...
-{
-    using log_api = irods::experimental::log::api;
-    if (_rsComm == nullptr || _dataObjInp == nullptr || _out_info == nullptr) {
+    if (nullptr == _comm || nullptr == _dataObjInp || nullptr == _out_info) {
         log_api::error("{}:{} Invalid input. Received one or more null pointers.", __func__, __LINE__);
         return SYS_NULL_INPUT;
     }
+
+    *_out_info = nullptr;
+
```
Comment (korydraughn) [inline server/api/src/rs_get_resource_info_for_operation.cpp:32]: This probably isn't necessary. Consider removing.

Comment (korydraughn) [reply server/api/src/rs_get_resource_info_for_operation.cpp:32]: This is fine. It guarantees the client will receive a nullptr on failure.

File: lib/api/include/irods/get_resource_info_for_operation.h (hunk 1 above)
Comment (korydraughn) [reply lib/api/include/irods/get_resource_info_for_operation.h]: Done.

**Comment by korydraughn** [comment]
Docs render well and all unit tests pass.

This is ready.

**Comment by korydraughn** [comment]
Had to add the unit test to the json list.

All unit tests pass.

File: server/api/src/rs_get_resource_info_for_operation.cpp [hunk 3]
```diff
@@ -1,34 +1,39 @@
...
 #include <fmt/format.h>
+#include <nlohmann/json.hpp>
 
+#include <algorithm>
 #include <array>
 #include <cstring> // for strdup
 #include <string>
 
-namespace
+int rs_get_resource_info_for_operation(RsComm* _comm, DataObjInp* _dataObjInp, char** _out_info)
```
Comment (alanking) [inline server/api/src/rs_get_resource_info_for_operation.cpp:22]: Might get dinged for trailing return type here

File: server/api/include/irods/rs_get_resource_info_for_operation.hpp [hunk 4]
```diff
@@ -8,36 +9,70 @@ struct RsComm;
...
-/// \endparblock
-/// \param[out] _out_info \parblock a JSON string with keys "host" and "resource_hierarchy", identifying target server and supplying a
-///                                 resource hierarchy string denoting a specific data object replica for the logical path given as input.
+///     - \p GET_RESOURCE_INFO_FOR_OPERATION_KW: The operation of interest. This option is required. The following
+///       values are supported:
+///         - \p "CREATE"
+///         - \p "WRITE"
+///         - \p "OPEN"
+///         - \p "UNLINK"
+///     - \p RESC_NAME_KW: Resource name hint. Given the name of a root resource, influences voting to select the
```
Comment (alanking) [inline server/api/include/irods/rs_get_resource_info_for_operation.hpp:22]: Maybe not worth mentioning, but I believe `DEST_RESC_NAME_KW` would also have a similar effect to `RESC_NAME_KW` for some or all operations.

**Review by alanking — COMMENTED** [review]
Seems good to me overall. Noticed a couple of things but they're negligible/possible ignorable.

File: server/api/src/rs_get_resource_info_for_operation.cpp (hunk 3 above)
Comment (korydraughn) [reply server/api/src/rs_get_resource_info_for_operation.cpp:22]: Oh right. Will wait for the clang-tidy report before changing.

File: server/api/include/irods/rs_get_resource_info_for_operation.hpp (hunk 4 above)
Comment (korydraughn) [reply server/api/include/irods/rs_get_resource_info_for_operation.hpp:22]: Ok. Here's a question, should we advertise that?

Comment (korydraughn) [reply server/api/include/irods/rs_get_resource_info_for_operation.hpp:22]: After some discussion, we've decided to not mention `DEST_RESC_NAME_KW`.

Users should stick to the documented options and values.

File: server/api/src/rs_get_resource_info_for_operation.cpp (hunk 3 above)
Comment (korydraughn) [reply server/api/src/rs_get_resource_info_for_operation.cpp:22]: Clang-tidy came back happy.

**Comment by korydraughn** [comment]
Let me know when we're ready to pound.

**Review by alanking — APPROVED** [review]
Huh, must be a warning now instead of an error.

Anyways, # it.

**Comment by korydraughn** [comment]
Yes, it was changed to a warning not too long ago.

Adding pounds.

**Comment by korydraughn** [comment]
#'d

**Pull request merged**